spec/services/model_importers/team_importer_spec.rb

```diff
@@ -0,0 +1,399 @@
+                )
+                expect(db_asset.version).to eq(
+                  json_asset['version']
+                )
+              end
+
+              # Tables
+              expect(db_step.step_tables.count).to eq(
+                json_step['step_tables'].count
+              )
+              expect(db_step.tables.count).to eq(
+                json_step['tables'].count
+              )
+
+              json_step['tables'].each do |json_table|
+                db_table = db_step.tables.find_by(
+                  contents: Base64.decode64(json_table['contents'])
+                )
+
+                # Basic fields
+                expect(db_table.created_at).to eq(
+                  json_table['created_at'].to_time
+                )
+                expect(db_table.updated_at).to eq(
+                  json_table['updated_at'].to_time
+                )
+                expect(db_table.created_by_id).to eq USER_ID
+                expect(db_table.last_modified_by_id).to eq USER_ID
+                expect(db_table.team_id). to eq @team.id
+
+                # Other fields
+                expect(db_table.name).to eq(json_table['name'])
+                expect(db_table.data_vector).to eq(
+                  Base64.decode64(json_table['data_vector'])
+                )
+              end
+
+              # Step comments
+              expect(db_step.step_comments.count).to eq(
+                json_step['step_comments'].count
+              )
+              json_step['step_comments'].each do |json_comment|
+                db_comment = db_step.step_comments.find_by(
+                  message: json_comment['message']
+                )
+
+                expect(db_comment.created_at).to eq(
+                  json_comment['created_at'].to_time
+                )
+                expect(db_comment.updated_at).to eq(
+                  json_comment['updated_at'].to_time
+                )
+                expect(db_comment.last_modified_by_id).to eq USER_ID
+              end
+            end
+
+            # Task comments
+            expect(db_module.task_comments.count).to eq(
+              my_module['task_comments'].count
+            )
+            my_module['task_comments'].each do |json_comment|
+              db_comment = db_module.task_comments.find_by(
+                message: json_comment['message']
+              )
+
+              expect(db_comment.created_at).to eq(
+                json_comment['created_at'].to_time
+              )
+              expect(db_comment.updated_at).to eq(
+                json_comment['updated_at'].to_time
+              )
+              expect(db_comment.last_modified_by_id).to eq USER_ID
+            end
+            #
+            # User assigns to the the module
+            unless my_module['user_my_modules'].empty?
+              expect(db_module.user_my_modules.first.user_id).to eq USER_ID
+            end
+          end
+        end
+      end
+    end
+  end
+end
```

Comment:
I bow to you for this exhaustive test.

Reply:
*bows back*. Could be much better though I believe :0